test/test_base58.py

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2018 John Newbery
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test base58.py"""
+import unittest
+
+import base58
+from binascii import unhexlify
+
+TEST_CASES = [
+    ["", ""],
+    ["61", "2g"],
+    ["626262", "a3gV"],
+    ["73696d706c792061206c6f6e6720737472696e67", "2cFupjhnEsSn59qHXstmK2ffpLv2"],
+    ["00CF00816E4BE7DBC1C3DF0C2B0BE2B77E4AD99A14111E6C6F", "1KsXUqbYDGz5gvBL2mo6as5KqKueRWc2Wr"],
+    ["00000000000000000000", "1111111111"],
+    # ["  1234", ""]
+]
+
+class Base58TestCase(unittest.TestCase):
+
+    def test_bytes_to_base58(self):
+        for exp_bin, exp_base58 in TEST_CASES:
```

Comment:
I'm not sure what `exp_` means here. Expected? In which case, I think I'd rename as follows:

`exp_bin` -> `hex`
`exp_bin` (line below) -> `bin`

Reply:
`hex` and `bin` are built-in function names, but I'll rename them to be `expected_hex` and `expected_bin` to be clear and descriptive.

Reply:
ah yes, of course.

Sorry, my previous comment wasn't clear. What I meant was that `exp_bin` isn't really 'expected' - it's provided. Perhaps `input_hex` and `input_bin` are better?